File: selfstudy/js/players_min.js
/*
task 63:
Players - Min
You need to find the minimum value among player heights in cm
*/

function lowest_players(heights) {
  return heights.length ? Math.min(...heights) : 0;
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: lowest_players([10, 20, 30]) },
  { input: [50, 60, 70], expected: lowest_players([50, 60, 70]) },
  { input: [1, 2, 3], expected: lowest_players([1, 2, 3]) },
  { input: [100, 200], expected: lowest_players([100, 200]) },
  { input: [], expected: lowest_players([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('lowest_players', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(lowest_players(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, lowest_players(t.input) === t.expected || JSON.stringify(lowest_players(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// done

File: selfstudy/golang/players_min.go
/*
task 263:
Players - Min
You need to find the minimum value among player heights in cm
*/

package main

import (
	"fmt"
	"reflect"
)

func LowestPlayers(heights []int) int {
	if len(heights) == 0 { return 0 }
	m := heights[0]
	for _, v := range heights {
		if v < m { m = v }
	}
	return m
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, LowestPlayers([]int{10, 20, 30})},
		{[]int{50, 60, 70}, LowestPlayers([]int{50, 60, 70})},
		{[]int{1, 2, 3}, LowestPlayers([]int{1, 2, 3})},
		{[]int{100, 200}, LowestPlayers([]int{100, 200})},
		{[]int{}, LowestPlayers([]int{})},
	}
	
	for idx, t := range tests {
		res := LowestPlayers(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done
